src/TodayEventMonitor.vala

```diff
@@ -0,0 +1,181 @@
+        foreach (var alarm_uid in event.get_alarm_uids ()) {
+            ECal.ComponentAlarm e_alarm = event.get_alarm (alarm_uid);
+            ECal.ComponentAlarmAction action;
+
+#if E_CAL_2_0
+            action = e_alarm.get_action ();
+#else
+            e_alarm.get_action (out action);
+#endif
+            switch (action) {
+                case (ECal.ComponentAlarmAction.DISPLAY):
+                    ECal.ComponentAlarmTrigger trigger;
+#if E_CAL_2_0
+                    trigger = e_alarm.get_trigger ();
+#else
+                    e_alarm.get_trigger (out trigger);
+#endif
+                    if (trigger.get_kind () == ECal.ComponentAlarmTriggerKind.RELATIVE_START) {
+                        ICal.Duration duration = trigger.get_duration ();
+                        var start_time = Calendar.Util.icaltime_to_datetime (comp.get_dtstart ());
+                        var now = new DateTime.now_local ();
+                        if (now.compare (start_time) > 0) {
+                            continue;
+                        }
+                        start_time = start_time.add_weeks (-(int)duration.get_weeks ()); //vala-lint=space-before-paren
+                        start_time = start_time.add_days (-(int)duration.get_days ()); //vala-lint=space-before-paren
+                        start_time = start_time.add_hours (-(int)duration.get_hours ()); //vala-lint=space-before-paren
+                        start_time = start_time.add_minutes (-(int)duration.get_minutes ()); //vala-lint=space-before-paren
+                        start_time = start_time.add_seconds (-(int)duration.get_seconds ()); //vala-lint=space-before-paren
+                        if (start_time.get_year () == now.get_year () && start_time.get_day_of_year () == now.get_day_of_year ()) {
+                            var time = time_until_now (start_time);
+                            if (time >= 0) {
+                                add_timeout.begin (source, event, (uint)time);
+                            }
+                        }
+                    }
+                    continue;
+                default:
+                    continue;
+            }
+        }
+    }
+
+    public async void add_timeout (E.Source source, ECal.Component event, uint interval) {
+        var uid = "%u-%u".printf (interval, GLib.Random.next_int ());
+        event_uids.set (event, uid);
+        debug ("adding timeout uid:%s", uid);
+        Timeout.add_seconds (interval, () => {
+            var extension = (E.SourceAlarms)source.get_extension (E.SOURCE_EXTENSION_ALARMS);
+            if (extension != null) {
+                extension.set_last_notified (new DateTime.now_local ().to_string ());
+            }
+
+            queue_event_notification (event, uid);
+            return false;
+        });
+    }
+
+    public void queue_event_notification (ECal.Component event, string uid) {
```

Comment:
```suggestion
    public void send_event_notification (ECal.Component event, string uid) {
```

I think this is a little clearer. The `add_timeout` function is really "queuing" the notification in the sense of adding it to a queue, while this function is called to actually send the notification.

Reply:
Renamed the function.